R/stat-.r

```diff
@@ -97,10 +97,18 @@ Stat <- ggproto("Stat",
     args <- c(list(data = quote(data), scales = quote(scales)), params)
     dapply(data, "PANEL", function(data) {
       scales <- layout$get_scales(data$PANEL[1])
-      try_fetch(do.call(self$compute_panel, args), error = function(cnd) {
-        cli::cli_warn("Computation failed in {.fn {snake_class(self)}}", parent = cnd)
-        new_data_frame()
-      })
+      try_fetch(do.call(self$compute_panel, args),
```

Comment:
Unless I'm missing something, I think this should be more like:

```R
try_fetch(do.call(self$compute_panel, args),
  rlib_error_package_not_found = function(cnd) {
    cli::cli_abort("Aborted computation in {.fn {snake_class(self)}}", parent = cnd)
  },
  error = function(cnd) {
    cli::cli_warn("Computation failed in {.fn {snake_class(self)}}", parent = cnd)
    new_data_frame()
  }
)
```

Reply:
Also note that the `in {.fn {snake_class(self)}}` is temporary unless we replace it with a proper `call`. Not sure if we want to do that now or later.

Reply:
Also note that the `in {.fn {snake_class(self)}}` is temporary unless we replace it with a proper `call`. Not sure if we want to do that now or later.

Reply:
This looks not bad to me, but let me think. I'm a noob at cli...

```
> ggplot(diamonds, aes(carat, price)) + geom_hex()
Error in `fun()` at ggplot2/R/compat-plyr.R:375:4:
! Aborted computation in `stat_binhex()`
Caused by error in `f()` at ggplot2/R/ggproto.r:178:11:
! The package `hexxbin` is required for `stat_binhex()`
Run `rlang::last_error()` to see where the error occurred.
```

Reply:
I think I'd structure the errors in this way, just to avoid multiple formulations of the same idea (an error occurred):

```
Error in `fun()` at ggplot2/R/compat-plyr.R:375:4:
Caused by error in `stat_binhex()`:
Caused by error in `f()` at ggplot2/R/ggproto.r:178:11:
! The package `hexxbin` is required for `stat_binhex()`
```

Reply:
> Caused by error in `stat_binhex()`:
> Caused by error in `f()` at ggplot2/R/ggproto.r:178:11:

I couldn't figure out how to structure like this. Can we add the additional "Caused by" like without providing the error message (the line starting with `!`)?